ConfigSpace/hyperparameters.pyx

```diff
@@ -1322,16 +1355,36 @@ cdef class CategoricalHyperparameter(Hyperparameter):
     def _sample(self, rs: np.random.RandomState, size: Optional[int]=None) -> Union[int, np.ndarray]:
         return rs.randint(0, self.num_choices, size=size)
 
-    def _transform(self, vector: Union[np.ndarray, float, int]) -> Optional[Union[str, int, float]]:
-        if not np.isfinite(vector):
-            return None
+    cpdef np.ndarray _transform_vector(self, np.ndarray vector):
+        if np.any(~np.isfinite(vector)):
+            raise ValueError('vector %s contains non-finite numbers' % vector)
+
         if np.equal(np.mod(vector, 1), 0):
-            return self.choices[int(vector)]
-        else:
-            raise ValueError('Can only index the choices of the categorical '
+            return self.choices(vector.astype(int))
```

Comment:
Why did you replace the way the choices array is accessed (via `[]`) by `()`?

Reply:
Hmm that's weird... Shouldn't this have given an error?

Reply:
Yes, I would have expected that, too.